tests/Simplify/EqualTest.elm

```diff
@@ -934,6 +934,787 @@ a = ({ a = 1 }).a == ({ a = 1 }).b
+"""
+                        ]
+        , test "should replace 0 /= Dict.size d with (not << Dict.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 /= Dict.size d
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << Dict.isEmpty)"
+                            , details =
+                                [ "Whereas Dict.size takes as long to run as the number of elements in the Dict,"
+                                , "Dict.isEmpty runs in constant time."
+                                ]
+                            , under = "0 /= Dict.size d"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (not << Dict.isEmpty) d
+"""
+                        ]
+        , test "should replace (d |> Dict.size) /= 0 with (not << Dict.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = (d |> Dict.size) /= 0
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << Dict.isEmpty)"
+                            , details =
+                                [ "Whereas Dict.size takes as long to run as the number of elements in the Dict,"
+                                , "Dict.isEmpty runs in constant time."
+                                ]
+                            , under = "(d |> Dict.size) /= 0"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (d |> (not << Dict.isEmpty))
+"""
+                        ]
+        , test "should replace 0 /= (d |> Dict.size) with (not << Dict.isEmpty)" <|
+            \() ->
+                """module A exposing (..)
+a : Bool
+a = 0 /= (d |> Dict.size)
+"""
+                    |> Review.Test.run ruleWithDefaults
+                    |> Review.Test.expectErrors
+                        [ Review.Test.error
+                            { message = "This can be replaced with a call to (not << Dict.isEmpty)"
+                            , details =
+                                [ "Whereas Dict.size takes as long to run as the number of elements in the Dict,"
+                                , "Dict.isEmpty runs in constant time."
+                                ]
+                            , under = "0 /= (d |> Dict.size)"
+                            }
+                            |> Review.Test.whenFixed """module A exposing (..)
+a : Bool
+a = (d |> (not << Dict.isEmpty))
+"""
+                        ]
+        ]
+
+
+arrayTests : Test
+arrayTests =
+    describe "Array.length should be Set.isEmpty"
```

Comment:
```suggestion
    describe "Array.length should be Array.isEmpty"
```

(Same issue for the dict tests at least)

Reply:
This pushes me more to rework the tests generally. These are very repetitive (obviously). I should instead rotate through the pairs of replaced/replacement functions inside of one list of tests. I should also vary the collection expressions and not just go replacing single letter/word expressions but rather something larger.

Reply:
I think it's fine the way you did it, errors like this are bound to happen, but once they're fixed (a nice IDE helps with that), the tests are really nice, even if they're repetitive. At least in my opinion, you're allowed to disagree :sweat_smile: